portfolio/src/main/webapp/script.js

```diff
@@ -154,12 +154,47 @@ const selectionChangeHandler = async (value) => {
   loadComments(value);
 }
 
+/**
+ * Get a specific GET parameter from the query string
+ * @param{String} name The name of the parameter
+ */
+const getParameter = (name) => {
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  return urlParams.get(name);
+}
+
+/**
+ * Set the value of the #maxComments element on the page
+ * @param{Number} value The value to be assigned
+ */
+const setMaxComments = (value) => {
+  const maxCommentsElement = document.getElementById('maxComments');
+  const selectedIndex = maxCommentsElement.selectedIndex;
+  const optionValues = [];
+  // constructing the optionValues[] array in this way is required as
+  // maxCommentsElement.options is an HTMLCollection object, and does not
+  // have a .indexOf() function
+  for (let option of maxCommentsElement.options) {
+    optionValues.push(option.value);
+  }
+  const index = optionValues.indexOf(value);
+  // if the passed value is not a valid option (not found in options array), set the value to 5
+  if (index === -1) {
+    // option with value 5 is stored at index 0
+    index = 0;
```

Comment:
You set index as a const, but if you hit this condition you then change index so it's not really a constant.

Reply:
my mistake, fixed 👍 https://github.com/thomasogara/step-repository/pull/27/commits/c006624b00cb551b6e54d27bfa80c91199b53284